VideoLocker/src/main/java/org/edx/mobile/view/adapters/CourseOutlineAdapter.java

```diff
@@ -379,52 +354,77 @@ private  View getRowViewForContainer(int position, View convertView, ViewGroup p
             holder.rowSubtitle.setText(component.getFormat());
         }
 
-        //support video download for video type
-        final int totalCount = component.getBlockCount().videoCount;
-        if (totalCount == 0 ){
+        // support video download for video type excluding the ones only viewable on web
+        int webOnlyCount = dbStore.getWebOnlyVideosCountBySection(courseId, chapterId, sequentialId, null);
+        final int totalDownloadableVideos = component.getBlockCount().videoCount - webOnlyCount;
+        if (totalDownloadableVideos == 0 ){
             holder.numOfVideoAndDownloadArea.setVisibility(View.GONE);
         } else {
+            holder.bulkDownload.setVisibility(View.VISIBLE);
             holder.noOfVideos.setVisibility(View.VISIBLE);
-            holder.noOfVideos.setText("" + totalCount);
-
-            if (  row.numOfVideoNotDownloaded == 0 ){
-                holder.bulkDownload.setVisibility(View.GONE);
+            holder.noOfVideos.setText("" + totalDownloadableVideos);
+
+            List<Long> downloadingList = dbStore.getDownloadingVideoDmIdsForSection(courseId, chapterId, sequentialId, null);
+            boolean areAllVideosDownloaded = dbStore.isVideoDownloadedInSection(courseId, chapterId, sequentialId, null);
```

Comment:
The `isVideoDownloadedInSection()` will return true if _any_ video in that section is downloading, not necessarily when _all_ of them are downloading. You need to look at the deleted section from line 393 on how to get that.